tools/codegen/model.py

```diff
@@ -1496,3 +1503,40 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+@dataclass(frozen=True)
+class PrecomputedElement:
+    name: str
+    ty: str
+
+    @staticmethod
+    def build(src: str) -> 'PrecomputedElement':
+        ty, name = src.split(" ")
+        return PrecomputedElement(name=name, ty=ty)
+
+    def decl(self) -> str:
+        return f"{self.ty} {self.name}"
+
+@dataclass(frozen=True)
+class Precompute:
+    elements: List[Argument]
+    replace: Optional[Dict[str, List[Argument]]]
```

Comment:
It seems simpler to just not make this Optional at all

Reply:
Also not sure why the target is Argument; there are no types in the substitution

Reply:
Ah yes, an empty map does also evaluate to `False`...